### PR TITLE
Start moving ontario govt data sources to classes with tests

### LINKED ISSUE #31: Migrate CSV logic to module

### LINKED ISSUE #32: Migrate ontario government datasource to module

### PR DESCRIPTION
**Pull request by waffle2k**

### CODE REVIEW

**Comment by tommertron** [comment]
I'm still a little out of my depths on classes and unit tests tbh, so I'm not sure what to review here. These are net new files, so I don't see an issue merging, but would be good to get a breakdown of the approach here at some point.

**Comment by waffle2k** [comment]
I published the PR without filling the description <facepalm>.

This is a WIP PR, and I'll be updating it over the next couple of days.  I just wanted some early feedback on the direction more than anything else.

The goal is to refactor some of the code out of the `covid_data_getter.py` so that it can be properly unit tested.  I'm doing this by making a new module for the relevant parts, and adding test coverage.  Once those things are completed, I'll cut the old code to the new code.

This will achieve a few things:
- make the codebase a little easier to read
- provide test coverage
- split up the code into logical modules so that it's easier to understand which parts of the codebase are related by keeping related code in modules together

I've started with the function to grab the raw JSON data from the ontario gov API.  Will attempt the same for the CSV data store later on.

**Comment by tommertron** [comment]
Thanks that makes sense!

**Comment by tommertron** [comment]
Wait so I should hold off on merging then right?

**Comment by waffle2k** [comment]
### Manual tests

## Create from fresh
```
(.venv)  pmb@dev  ~/github/CovidReport   querytests ±  rm foo.csv
(.venv)  pmb@dev  ~/github/CovidReport   querytests ±  python datasources/ontgov.py
(.venv)  pmb@dev  ~/github/CovidReport   querytests ±  cat foo.csv                 
date,total_doses_administered,total_individuals_at_least_one,total_individuals_fully_vaccinated,total_individuals_3doses,Total Cases,Number of patients hospitalized with COVID-19,Number of patients in ICU due to COVID-19
2022-01-12,28853124,12307610,11493087,5033258,906031,3448,505
2022-01-11,28693247,12296949,11482911,4894431,896248,3220,477
2022-01-10,28551554,12288683,11474711,4769378,888297,2467,438
2022-01-09,28457813,12282507,11469856,4686745,878591,2419,412
2022-01-08,28324152,12271504,11462390,4571616,866632,2594,385
2022-01-07,28140051,12260375,11454222,4406948,853270,2472,338
2022-01-06,27945958,12249723,11445183,4232672,841371,2279,319
2022-01-05,27750953,12239815,11436474,4056554,828032,2081,288
2022-01-04,27570940,12230179,11427792,3895067,816450,1290,266
2022-01-03,27422363,12222703,11421633,3760163,805098,1232,248
```

## Remove a line in the middle and re-run, and show that the line was backfilled
```
diff -Naur foo.csv foo_line_removed.csv
--- foo.csv     2022-01-12 20:00:43.048975346 -0800
+++ foo_line_removed.csv        2022-01-12 20:02:54.243625442 -0800
@@ -3,7 +3,6 @@
 2022-01-11,28693247,12296949,11482911,4894431,896248,3220,477
 2022-01-10,28551554,12288683,11474711,4769378,888297,2467,438
 2022-01-09,28457813,12282507,11469856,4686745,878591,2419,412
-2022-01-08,28324152,12271504,11462390,4571616,866632,2594,385
 2022-01-07,28140051,12260375,11454222,4406948,853270,2472,338
 2022-01-06,27945958,12249723,11445183,4232672,841371,2279,319
 2022-01-05,27750953,12239815,11436474,4056554,828032,2081,288
```
## And run it
```
(.venv)  ✘ pmb@dev  ~/github/CovidReport   querytests ±  cp foo_line_removed.csv foo.csv 
(.venv)  pmb@dev  ~/github/CovidReport   querytests ±  python datasources/ontgov.py 
(.venv)  ✘ pmb@dev  ~/github/CovidReport   querytests ±  diff -Naur foo_line_removed.csv foo.csv 
--- foo_line_removed.csv        2022-01-12 20:02:54.243625442 -0800
+++ foo.csv     2022-01-12 20:03:59.291443581 -0800
@@ -3,6 +3,7 @@
 2022-01-11,28693247,12296949,11482911,4894431,896248,3220,477
 2022-01-10,28551554,12288683,11474711,4769378,888297,2467,438
 2022-01-09,28457813,12282507,11469856,4686745,878591,2419,412
+2022-01-08,28324152,12271504,11462390,4571616,866632,2594,385
 2022-01-07,28140051,12260375,11454222,4406948,853270,2472,338
 2022-01-06,27945958,12249723,11445183,4232672,841371,2279,319
 2022-01-05,27750953,12239815,11436474,4056554,828032,2081,288
```

File: datasources/ontgov.py [hunk 1]
```diff
@@ -1,6 +1,139 @@
...
+        return [DateRanges._daysago(i) for i in range(size)]
+
+    @staticmethod
+    def _daysago(days: int) -> datetime:
+        return date.today() - timedelta(days=days)
+
+
+class CovidData:
+    def missing(self, date: datetime) -> bool:
+        return True
```
Comment (waffle2k) [inline datasources/ontgov.py]: Uh, this should be trimmed.

**Pull request merged**